src/visual.ts

```diff
@@ -1278,6 +1278,82 @@ module powerbi.extensibility.visual {
                 .exit()
                 .remove();
 
+            // create labels on link as A - B : Value
+            let linkTextData: SankeyDiagramLink[] = sankeyDiagramDataView.links.filter((link: SankeyDiagramLink) => {
+                return link.height > SankeyDiagram.MinSize && this.dataView.settings.linkLabels.show;
+            });
+            // add defs element to svg
+            let svgDefs = this.root
+                .selectAll("defs");
+
+            let svgDefsSelection = svgDefs.data([1]);
+            svgDefsSelection
+                .enter()
+                .append("defs");
+
+            svgDefsSelection
+                .exit()
+                .remove();
+
+            let singleDefsElement = d3.select(svgDefsSelection.node());
+
+            // add text path for lables
+            let linkLabelPaths = singleDefsElement.selectAll(".linkLabelPaths");
+
+            let linkLabelPathsSelection = linkLabelPaths.data(linkTextData);
+
+            linkLabelPathsSelection
+                .enter()
+                .append("path")
+                .classed("linkLabelPaths", true);
+
+            linkLabelPathsSelection
+                .attr({
+                    d: (link: SankeyDiagramLink) => {
+                        return this.getSvgPath(link);
+                    },
+                    id: (link: SankeyDiagramLink) => `${(link.source.label.internalName || "").replace(/\W*/g,"")}-${(link.destination.label.internalName || "").replace(/\W*/g,"")}`
+                });
+
+            linkLabelPathsSelection
+                .exit()
+                .remove();
+
+            // add text by using paths from defs
+            let linkLabelTexts = this.main
+                .select(SankeyDiagram.LinksSelector.selector)
+                .selectAll(".linkLabelTexts");
+
+            let linkLabelTextSelection = linkLabelTexts.data(linkTextData);
+
+            linkLabelTextSelection
+                .enter()
+                .append("text")
+                .attr({
+                    "text-anchor": "middle"
+                })
+                .classed("linkLabelTexts", true);
+
+            linkLabelTextSelection.selectAll("textPath").remove();
```

Comment:
Can we use data-enter-append-remove sequence in order to eliminate removing DOM elements?